Managers/SceneMan.h

```diff
@@ -1176,9 +1176,22 @@ class SceneMan : public Singleton<SceneMan>, public Serializable {
 //                  all Air, then 0 is returned (Air's strength value is 0).
 
     // We use two accessors instead of default parameters, for lua compat
-    float CastMaxStrengthRay(const Vector &start, const Vector &end, int skip, unsigned char ignoreMaterial);
+    float CastMaxStrengthRay(const Vector& start, const Vector& end, int skip, unsigned char ignoreMaterial);
     float CastMaxStrengthRay(const Vector &start, const Vector &end, int skip) { return CastMaxStrengthRay(start, end, skip, g_MaterialDoor); };
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// Method:          CastMaxStrengthRayMaterial
+//////////////////////////////////////////////////////////////////////////////////////////
+// Description:     Traces along a vector and returns the strongest of all encountered pixels' materials
+//                  This will take wrapping into account.
+// Arguments:       The starting position.
+//                  The ending position.
+//                  For every pixel checked along the line, how many to skip between them
+//                  for optimization reasons. 0 = every pixel is checked.
+//                  A material ID to ignore, IN ADDITION to Air. This defaults to doors, for legacy script purposes
+// Return value:    The strongest material encountered
+    const Material * CastMaxStrengthRayMaterial(const Vector& start, const Vector& end, int skip, unsigned char ignoreMaterial);
```

Comment:
Wanna add lua access (and changelog entry) for this? Or do you figure it's not useful for lua stuff (I can see that argument, it's pretty particular)?

Reply:
Fair enough, I figured that might be the reason.

And yeah, CC has a trend of exposing a lot of stuff to its lua API, some of which probably shouldn't be. I can understand you not wanting to add another.